server/keyspace_service.go

```diff
@@ -82,16 +84,24 @@ func (s *KeyspaceServer) WatchKeyspaces(request *keyspacepb.WatchKeyspacesReques
 	ctx, cancel := context.WithCancel(s.Context())
 	defer cancel()
 
-	err := s.sendAllKeyspaceMeta(ctx, stream)
+	revision, err := s.sendAllKeyspaceMeta(ctx, stream)
 	if err != nil {
 		return err
 	}
-	watchChan := s.client.Watch(ctx, path.Join(s.rootPath, endpoint.KeyspaceMetaPrefix()), clientv3.WithPrefix())
+
 	for {
+		watchChan := s.client.Watch(ctx, path.Join(s.rootPath, endpoint.KeyspaceMetaPrefix()), clientv3.WithPrefix(), clientv3.WithRev(revision))
```

Comment:
I suspect this implementation of watch (w/ revision) isn't efficient:
1. if watchChan isn't closed/cancelled or doesn't return error, we should continue reusing watchChan instead of calling s.client.Watch after watchChan returning PUT/DETELE event every time.
2. When calling s.client.Watch, it shouldn't use the revision get from line 87. The revision should be updated with res.Revision around line 105 - 107.

Reply:
Oh, right. I intend to use for range.